drivers/hmis/app/models/ac_hmis/scoring/rule.rb

```
@@ -0,0 +1,119 @@
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module AcHmis
+  module Scoring
+    class Rule < ::GrdaWarehouseBase
+      self.table_name = 'hmis_scoring_rules'
+
+      RANGE = 'range'
+      EXACT_MATCH = 'exact_match'
+      VALUE = 'value'
+      INCLUDE = 'include'
+      CRITERIA_TYPES = [RANGE, EXACT_MATCH, VALUE, INCLUDE].freeze
+
+      validates :link_id, :form_definition_identifier, :algorithm, :criteria_type, :weight, presence: true
+      validates :criteria_type, inclusion: { in: CRITERIA_TYPES }
+      validate :valid_criteria_config
+
+      scope :for_algorithm, ->(algorithm) { where(algorithm: algorithm) }
+      scope :for_form, ->(form_definition_identifier) { where(form_definition_identifier: form_definition_identifier) }
+
+      # Evaluate this rule against a response value and return the score contribution
+      def evaluate(response_value)
+        weighted_value = 0
+
+        weighted_value = 1 if criteria_type == RANGE && range_match?(response_value)
+        weighted_value = 1 if criteria_type == EXACT_MATCH && exact_match?(response_value)
+        weighted_value = 1 if criteria_type == INCLUDE && include?(response_value)
+        weighted_value = convert_to_numeric(response_value) || 0 if criteria_type == VALUE
+
+        weighted_value * weight
+      end
+
+      private
+
+      def range_match?(value)
+        numeric_value = convert_to_numeric(value)
+        return false if numeric_value.nil?
+
+        # gather up all the range-related criteria in a list and check if they all match.
+        criteria_config.all? do |operator, threshold|
+          case operator
+          when 'gt' then numeric_value > threshold
+          when 'gte' then numeric_value >= threshold
+          when 'lt' then numeric_value < threshold
+          when 'lte' then numeric_value <= threshold
+          else true # ignore unknown operators
```

suggest raising here so we don't silently ignore misconfigurations. This would be unexpected since we have validations on this shape